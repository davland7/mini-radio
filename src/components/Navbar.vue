<template>
  <div class="navbar">
    <div class="navbar__logo">
      <img :src="require('@/images/logo.png')" alt="" />
      {{ messages.name }}
    </div>
    <div class="navbar__tabs">
      <div @click="tab(index)" @keyup.enter="tab(index)" v-for="(tabNumber, index) in totalTabs" :key="index" class="navbar__tab" :class="{'navbar__tab--current': currentTab === index}" tabindex="0" role="button">
        <span class="navbar__list">{{ messages.list }}</span> {{ tabNumber }}
      </div>
    </div>
  </div>
</template>

<script>
  export default {
    name: 'Tabs',
    props: [
      'currentTab',
      'totalTabs',
      'stationsPerTab'
    ],
    data () {
      return {
        messages: {
          name: chrome.i18n.getMessage('name'),
          list: chrome.i18n.getMessage('list')
        }
      }
    },
    methods: {
      tab (index) {
        this.$emit('tabClick', index)
      }
    }
  }
</script>

<style lang="scss">
  .navbar {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: 4px 0;

    &__logo {
      display: flex;
      align-items: center;
      height: 32px;
      font-size: 14px;
      font-weight: 600;
      color: var(--dark-color);

      img {
        padding: 0 6px;
        width: 24px;
        height: 24px;
      }
    }

    &__list {
      border: 0;
      clip: rect(0 0 0 0);
      height: 1px;
      margin: -1px;
      overflow: hidden;
      padding: 0;
      position: absolute;
      width: 1px;
    }

    &__tabs {
      display: flex;
    }

    &__tab {
      display: flex;
      justify-content: center;
      align-items: center;
      margin: 0 2px;
      width: 30px;
      height: 30px;
      font-size: 18px;
      color: var(--light-color);
      border: solid 1px var(--border-color);
      border-radius: 30px;
      cursor: pointer;

      &--current,
      &:hover {
        background-color: var(--content-color);;
      }
    }
  }
</style>
