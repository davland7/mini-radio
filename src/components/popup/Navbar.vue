<template>
  <div class="navbar">
    <div class="navbar__logo">
      <img src="images/icon24.png" />
      {{ messages.name }}
    </div>
    <div class="navbar__tabs">
      <div @click="tab(index)" v-for="(tabNumber, index) in totalTabs" class="navbar__tab" :class="{'navbar__tab--current': currentTab === index}">
          {{ tabNumber }}
      </div>
    </div>
  </div>
</template>

<script>
  export default {
    name: 'Tabs',
    props: [
      'currentTab',
      'totalTabs',
      'stationsPerTab'
    ],
    data () {
      return {
        messages: {
          name: chrome.i18n.getMessage('name')
        }
      }
    },
    methods: {
      tab (index) {
        this.$emit('tabClick', index)
      }
    }
  }
</script>

<style lang="scss">
  .navbar {
    display: flex;
    justify-content: space-between;
    align-items: center;
    margin: 4px 0;
  }

  .navbar__logo {
    display: flex;
    align-items: center;
    height: 32px;
    font-size: 14px;
    font-weight: 600;
    color: #3f3f3f;

    img {
      padding: 0 6px;
    }
  }

  .navbar__tabs {
    display: flex;
  }

  .navbar__tab {
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 0 2px;
    width: 30px;
    height: 30px;
    font-size: 18px;
    color: #363638;
    border: solid 1px #dcdcdd;
    cursor: pointer;

    &--current,
    &:hover {
      background-color: #e9e9ea;
    }
  }
</style>
