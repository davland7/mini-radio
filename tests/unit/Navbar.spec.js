import { shallowMount } from '@vue/test-utils'
import Navbar from '@/components/Navbar.vue'

const props = {
  messages: {
    list: 'List',
    name: 'Mini Radio'
  },
  currentTab: 1,
  totalTabs: 3,
  stationsPerTab: 5
}

describe('Navbar.vue', () => {
  it('renders correctly', () => {
    const wrapper = shallowMount(Navbar, { props })
    expect(wrapper.element).toMatchSnapshot()
  })
})
