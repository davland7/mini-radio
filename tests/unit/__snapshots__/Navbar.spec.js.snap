// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar.vue renders correctly 1`] = `
<div
  class="navbar"
>
  <div
    class="navbar__logo"
  >
    <img
      alt=""
      src=""
    />
     Mini Radio
  </div>
  <div
    class="navbar__tabs"
  >
    
    <div
      class="navbar__tab"
      role="button"
      tabindex="0"
    >
      <span
        class="navbar__list"
      >
        List
      </span>
       1
    </div>
    <div
      class="navbar__tab navbar__tab--current"
      role="button"
      tabindex="0"
    >
      <span
        class="navbar__list"
      >
        List
      </span>
       2
    </div>
    <div
      class="navbar__tab"
      role="button"
      tabindex="0"
    >
      <span
        class="navbar__list"
      >
        List
      </span>
       3
    </div>
    
  </div>
</div>
`;
